system-addon/lib/ShortURL.jsm

```diff
@@ -39,7 +54,7 @@ this.shortURL = function shortURL(link) {
   if (!link.url && !link.hostname) {
     return "";
   }
-  const {eTLD} = link;
+  const eTLD = link.eTLD || getETLD(link.url);
   const hostname = (link.hostname || new URL(link.url).hostname).replace(/^www\./i, "");
 
   // Remove the eTLD (e.g., com, net) and the preceding period from the hostname
```

Comment:
The following line for defaulting to .com shouldn't be necessary now that we're getting the actual eTLD always. `|| (hostname.match(/\.com$/) && 3);`